api/external_auth.proto

```diff
@@ -0,0 +1,152 @@
+syntax = "proto3";
+
+// [#proto-status: draft]
+
+package envoy.api.v2;
+
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+import "google/rpc/status.proto";
+
+// Authorization service.
+// Primarily responds with OK or NOT OK.
+service Authorization {
+   rpc Check(CheckRequest) returns (CheckResponse);
+}
+
+message CheckRequest {
+    // The name of the tenant for multi-tenant support.
+    string tenant_name = 1;
```

Comment:
Because Envoy can be deployed in different contexts, such as Istio mesh or Kubernetes cluster or GCE/EC2 network, there won't be a good term for it. That is why Google APIs simply use the term "name" and let people decide what it is, and document it as "The name of the entity this check is performed against."